runcomp.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+read -p "Defend Plus X (0 or 1): " defendPlusX
+read -p "Vision Channel (1,2 or Full): " visionChannel
+read -p "Specify Playbook file: " playbook
+
+make
+
+until ./run/soccer -pbk $playbook -defend $defendPlusX -vision $visionChannel; do
```

Comment:
Not a full review, but if you want to add default values to this (ex: user just hits enter), you can change your variable references to:

${playbook:-"DefaultValue"} instead of $playbook.

Keep in mind, if playbook has any value, including an invalid one (like a single letter), this won't trigger.